modules/gapi/src/backends/fluid/gfluidimgproc.cpp

```diff
@@ -1980,11 +1990,38 @@ struct Mapper {
         return u;
     }
 };
+}  // namespace linear
+
+namespace linear32f {
+struct Mapper {
+    typedef float alpha_type;
+    typedef int   index_type;
+    constexpr static const float unity = 1;
+
+    typedef MapperUnit<float, int> Unit;
+
+    static inline Unit map(double ratio, int start, int max, int outCoord) {
+        float f = static_cast<float>((outCoord + 0.5) * ratio - 0.5);
+        int s = cvFloor(f);
+        f -= s;
+
+        Unit u;
+
+        u.index0 = std::max(s - start, 0);
```

Comment:
is negative allowed?

Reply:
if result `s-start` is negative, std::max will choose `0`